app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1681,6 +1683,99 @@ impl GraphEditorModel {
         find_free_place(starting_from, direction, node_areas)
     }
 
+    pub fn new_node_position_after_dropping_edge(
+        &self,
+        edge_id: EdgeId,
+        mouse_position: Vector2,
+    ) -> Vector2 {
+        let edge_source_node_id = self.edge_source_node_id(edge_id);
+        let edge_source_node = edge_source_node_id.and_then(|id| self.nodes.get_cloned_ref(&id));
+        self.new_node_position_based_on_mouse(mouse_position, edge_source_node)
+    }
+
+    pub fn new_node_position_based_on_mouse_without_source(
+        &self,
+        mouse_position: Vector2,
+    ) -> Vector2 {
+        let nearest_node = self.find_nearest_node(mouse_position);
+        self.new_node_position_based_on_mouse(mouse_position, nearest_node)
+    }
+
+    fn find_nearest_node(&self, mouse_position: Vector2) -> Option<Node> {
```

Comment:
This function does not have documentation. It is not clear by reading its name what "nearest" means.